File: aat-android/src/main/java/ch/bailu/aat/views/preferences/ConnectToSensorsView.java
package ch.bailu.aat.views.preferences;

import android.view.View;

import ch.bailu.aat.R;
import ch.bailu.aat.services.InsideContext;
import ch.bailu.aat.services.ServiceContext;
import ch.bailu.aat_lib.service.sensor.SensorState;
import ch.bailu.aat.util.ui.UiTheme;
import ch.bailu.aat.views.LabelTextView;
import ch.bailu.aat_lib.dispatcher.OnContentUpdatedInterface;
import ch.bailu.aat_lib.gpx.GpxInformation;
import ch.bailu.aat_lib.gpx.InfoID;
import ch.bailu.aat_lib.gpx.StateID;

public class ConnectToSensorsView extends LabelTextView implements View.OnClickListener, OnContentUpdatedInterface {

    private String busy="";
    private final ServiceContext scontext;

    public ConnectToSensorsView(ServiceContext s, UiTheme theme) {
        super(s.getContext(), s.getContext().getString(R.string.sensor_connect), theme);
        scontext = s;
        setText();
        setOnClickListener(this);
        theme.button(this);
    }


    private void setText() {
        setText(SensorState.getOverviewString() + " " + busy);
    }


    @Override
    public void onClick(View v) {
        new InsideContext(scontext) {
            @Override
            public void run() {
                scontext.getSensorService().updateConnections();
            }
        };
    }

    @Override
    public void onContentUpdated(int iid, GpxInformation info) {
        if (iid == InfoID.SENSORS) {
            if (info.getState() == StateID.WAIT)
                busy = getContext().getString(R.string.gps_wait);
            else busy = "";

            setText();
        }

    }
}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/preferences/ConnectToSensorsView.kt
package ch.bailu.aat.views.preferences

import android.view.View
import ch.bailu.aat.R
import ch.bailu.aat.services.ServiceContext
import ch.bailu.aat.util.ui.theme.UiTheme
import ch.bailu.aat.views.layout.LabelTextView
import ch.bailu.aat_lib.dispatcher.TargetInterface
import ch.bailu.aat_lib.gpx.GpxInformation
import ch.bailu.aat_lib.gpx.InfoID
import ch.bailu.aat_lib.gpx.StateID
import ch.bailu.aat_lib.service.sensor.SensorState

class ConnectToSensorsView(private val scontext: ServiceContext, theme: UiTheme) : LabelTextView(
    scontext.getContext(), scontext.getContext().getString(R.string.sensor_connect), theme
), View.OnClickListener, TargetInterface {
    private var busy = ""

    init {
        setText()
        setOnClickListener(this)
        theme.button(this)
    }

    private fun setText() {
        setText("${SensorState.overviewString} $busy")
    }

    override fun onClick(v: View) {
        scontext.insideContext { scontext.getSensorService().updateConnections() }
    }

    override fun onContentUpdated(iid: Int, info: GpxInformation) {
        if (iid == InfoID.SENSORS) {
            busy = if (info.getState() == StateID.WAIT) context.getString(R.string.gps_wait) else ""
            setText()
        }
    }
}
